diff --git a/lib/python/Plugins/SystemPlugins/WirelessLan/Wlan.py b/lib/python/Plugins/SystemPlugins/WirelessLan/Wlan.py
index a9ff078bb..239df62ce 100644
--- a/lib/python/Plugins/SystemPlugins/WirelessLan/Wlan.py
+++ b/lib/python/Plugins/SystemPlugins/WirelessLan/Wlan.py
@@ -56,8 +56,9 @@ class Wlan:
 
 		self.asciitrans = str.maketrans(a, b)
 
-	def asciify(self, str):
-		return str.translate(self.asciitrans)
+	def asciify(self, str_):
+		print("ASCII IS: ", self.asciitrans)
+		return str_.translate(self.asciitrans)
 
 	def getWirelessInterfaces(self):
 		return getWNICnames()
@@ -80,11 +81,12 @@ class Wlan:
 
 		ifobj = Wireless(self.iface) # a Wireless NIC Object
 
-		try:
+		if 1:
 			scanresults = ifobj.scan()
-		except:
-			scanresults = None
-			print("[Wlan.py] No wireless networks could be found")
+			print("SCAN RESULTS IS: ", scanresults)
+		#except:
+		#	scanresults = None
+		#	print("[Wlan.py] No wireless networks could be found")
 		aps = {}
 		if scanresults is not None:
 			(num_channels, frequencies) = ifobj.getChannelInfo()
@@ -108,8 +110,8 @@ class Wlan:
 
 				extra = []
 				for element in result.custom:
-					element = element.encode()
-					extra.append(strip(self.asciify(element)))
+					element = element.decode()
+					extra.append(str.strip(self.asciify(element)))
 				for element in extra:
 					if 'SignalStrength' in element:
 						signal = element[element.index('SignalStrength') + 15:element.index(',L')]
@@ -127,7 +129,7 @@ class Wlan:
 					'bssid': result.bssid,
 					'channel': channel,
 					'encrypted': encryption,
-					'essid': strip(self.asciify(result.essid)),
+					'essid': str.strip(self.asciify(result.essid)),
 					'iface': self.iface,
 					'maxrate': ifobj._formatBitrate(result.rate[-1][-1]),
 					'noise': '',#result.quality.nlevel-0x100,
